cmd/vulcan-aws-trusted-advisor/main.go

```diff
@@ -148,12 +152,48 @@ func scanAccount(opt options, target, assetType string, logger *logrus.Entry, st
 			enqueued++
 		}
 	}
-	// Let's chill a bit before getting results
+
+	// If some check was enqueued for refreshing
+	// poll it's status and wait up until opt.RefreshTimeout
 	if enqueued > 0 {
-		logger.Infof("waiting for checks to be refreshed. Sleeping for %d seconds ...", opt.RefreshTimeout)
-		time.Sleep(time.Duration(opt.RefreshTimeout) * time.Second)
+		t := time.NewTicker(time.Duration(opt.RefreshTimeout) * time.Second)
+
+	LOOP:
+		for {
+			select {
+			case <-t.C:
+				break LOOP
+			default:
+				checkStatus, err := s.DescribeTrustedAdvisorCheckRefreshStatuses(
+					&support.DescribeTrustedAdvisorCheckRefreshStatusesInput{
+						CheckIds: checkIds,
+					},
+				)
+				if err != nil {
+					if awsErr, ok := err.(awserr.Error); ok {
+						if awsErr.Code() != "InvalidParameterValueException" {
+							return err
```

Comment:
If we exit here the `time.Ticker` declared on line 159 will continue running?

Reply:
If we exit here the ticker will be "deallocated" no? Because we exit the context on which the ticker was declared, no?

Reply:
Change it for a defer stop of the ticker right after initialization, it won't be stopped explicitly after the LOOP, but it will before exiting the function which should guarantee we don't leak it.

Reply:
Perfect! Thanks, @ka3de!